Merging Playlist Manager Feature

anishsudini opened the pull request:
All back-end and API logic complete. Preset creation logic, and playlist dynamic reordering logic complete and tested. 
Delay issue fixed. 
Only issue left pertains to UI of the feature and permanently effecting the Spotify App instead of for just one instance on load.

anishsudini committed:
Initial Logic For Playlist Manager

Implemented getting and locally cache/storing a playlist's tracks' songIDs and those tracks' audio features as well.

Created three files so far, 2 in the services folder for handling API requests and response structure, and 1 in the components folder handling getting the playlist ID of an owned playlist on a user's screen.

anishsudini committed:
Merge pull request #14 from rrk0804/main

Pulling down changes for testing

anishsudini committed:
Playlist Manager Feature

Preset creation logic, and playlist dynamic reordering logic complete and tested. Delay issue fixed. Only issue left pertains to UI of the feature and permanently effecting the Spotify App instead of for just one instance on load.

anishsudini committed:
Fix for Playlist Manager UI Injection

anishsudini committed:
Implemented Fixes for CSS Bugs & Functionality Bugs/Flow Issues

Modified code to include mutation observers for various aspects of the page to determine and implement permanent UI injections. Discuss style with the team, made some changes to meet overall agreeance.

anishsudini committed:
Small Fixes for TypeScript Errors

Fixed some TypeScript errors (that were not affecting code functionality) but just needed to add some error handling for them.

rrk0804 commented:
The UI looks great. However, there is a delay reordering tracks using musical characteristics

anishsudini commented:
Found issue for delay: root cause is due to not having "Custom Order" selected in the dropdown sorting options. Implementing fix to mimic this programmatically to avoid user interaction or issue.

anishsudini committed:
Fixed Delay/Custom Order Issue

Also removed Duration as a custom sorting option since it already is provided by Spotify.

anishsudini commented:
Implemented fix for above issue^^.

skycube19 reviewed (approved):
Briefly tested it out locally and it seems to work as intended. Great job!

rrk0804 commented:
New fix works on my machine. Great job!
